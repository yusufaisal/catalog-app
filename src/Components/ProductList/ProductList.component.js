import * as React from 'react';
import { View, FlatList, ScrollView } from 'react-native';

import ProductItem from '../ProductItem/ProductItem.component';

import styles from './ProductList.component.styles';

const _renderProductItem = ({ item }) => (
  <ProductItem {...item}/>
);

const _renderList = (props) => (
  <FlatList
    data={props.products}
    renderItem={_renderProductItem}
    numColumns={2}
    keyExtractor={(item, index) => item.id.toString()}
    showsVerticalScrollIndicator={false}
  />
)

const ProductList = (props) => (
  <View style={styles.container}>
    {_renderList(props)}
  </View>
);

export default ProductList;