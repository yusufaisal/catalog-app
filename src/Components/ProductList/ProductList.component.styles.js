const shadow = {
  shadowColor: '#171717',
  shadowOffset: {width: 1, height: 1},
  shadowOpacity: 0.1,
  shadowRadius: 3,
}

export default {
  container: {
    flex: 1,
    marginVertical: 10,
    marginHorizontal: 20,
  }
};