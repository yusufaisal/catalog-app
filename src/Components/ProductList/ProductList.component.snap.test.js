import React from 'react';
import renderer from 'react-test-renderer';

import ProductList from './ProductList.component';

jest
  .mock('../ProductItem/ProductItem.component', () => 'ProductItem');
  
describe('ProductList Component', () => {
  const products = [{
    "id": "1",
    "image": "https://fe.dev.dxtr.asia/storage/products/Product_07.png",
    "name": "Pumma Runing Shoes",
    "price": "$350",
    "off": "20% Off",
    "rating": 4,
    "new": true,
    "isFav": true,
    "out_of_stock": false,
    "reviewCount": 150,
  }, {
    "id": "2",
    "image": "https://fe.dev.dxtr.asia/storage/products/Product_01.png",
    "name": "Heels Sandals",
    "price": "$450",
    "off": "",
    "rating": 4.5,
    "new": false,
    "isFav": false,
    "out_of_stock": true,
  }]

  it('should render ProductItem properly', () => {
    const tree = renderer
      .create(<ProductList products={products}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});