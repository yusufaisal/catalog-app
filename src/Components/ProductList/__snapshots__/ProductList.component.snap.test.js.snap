// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList Component should render ProductItem properly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginHorizontal": 20,
      "marginVertical": 10,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": "1",
          "image": "https://fe.dev.dxtr.asia/storage/products/Product_07.png",
          "isFav": true,
          "name": "Pumma Runing Shoes",
          "new": true,
          "off": "20% Off",
          "out_of_stock": false,
          "price": "$350",
          "rating": 4,
          "reviewCount": 150,
        },
        Object {
          "id": "2",
          "image": "https://fe.dev.dxtr.asia/storage/products/Product_01.png",
          "isFav": false,
          "name": "Heels Sandals",
          "new": false,
          "off": "",
          "out_of_stock": true,
          "price": "$450",
          "rating": 4.5,
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <ProductItem
            id="1"
            image="https://fe.dev.dxtr.asia/storage/products/Product_07.png"
            isFav={true}
            name="Pumma Runing Shoes"
            new={true}
            off="20% Off"
            out_of_stock={false}
            price="$350"
            rating={4}
            reviewCount={150}
          />
          <ProductItem
            id="2"
            image="https://fe.dev.dxtr.asia/storage/products/Product_01.png"
            isFav={false}
            name="Heels Sandals"
            new={false}
            off=""
            out_of_stock={true}
            price="$450"
            rating={4.5}
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
